perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -3762,10 +3765,60 @@ public void updateExistingMemberWhileSynchronization(PerunSession sess, Group gr
 		//Synchronize userExtSources (add not existing)
 		addUserExtSources(sess, candidate, memberToUpdate);
 
+		if (isGroupMember(sess, group, memberToUpdate)) {
+			synchronizeGroupStatus(sess, memberToUpdate, candidate.getExpectedSyncGroupStatus(), group);
+		}
+
 		//Set correct member Status
 		updateMemberStatus(sess, memberToUpdate);
 	}
 
+	/**
+	 * Update a group member status based on the status in candidate object.
+	 * The member-group expiration attribute is set to now when expiring a member
+	 * and updated based on the group policy if validating a member.
+	 *
+	 * @param sess perun session
+	 * @param memberToSynchronize member being synchronized
+	 * @param newStatusString the group status from the extSource
+	 * @param group the group being synchronized
+	 */
+	private void synchronizeGroupStatus(PerunSession sess, Member memberToSynchronize, String newStatusString, Group group) {
+		MemberGroupStatus previousStatus = getDirectMemberGroupStatus(sess, memberToSynchronize, group);
+		MemberGroupStatus newStatus;
+		if (newStatusString == null) {
+			// default policy is VALID
+			log.warn("Missing status column while synchronizing group {}.", group.getName());
```

Comment:
Most synchronizations will not use the group status, therefore this message would just spam the logs. It would be probably better to remove it.